### PR TITLE
Fix unit test not running on modules issue

### LINKED ISSUE #71: Codelens actions don't work for Proofs inside Modules
The codelens trigger for running or debugging a unit test inside modules don't work as they need to have extra arguments present for the module and filename. They should be added when the proof is inside a module.

### PR DESCRIPTION
**Pull request by jaisnan**
### Description of changes: 

Adds module and file name to the metadata on the harnesses collected, which in turn can be passed to the unit test runner and debug connection. 

### Resolved issues:

Resolves #71 

### Testing:

* How is this change tested? Unit test

* Is this a refactor change?

### Checklist
- [x] Each commit message has a non-empty body, explaining why the change was made
- [x] Methods or procedures are documented
- [x] Regression or unit tests are included, or existing tests cover the modified code
- [x] My PR is restricted to a single feature or bugfix

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 and MIT licenses.

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
   "name": "kani-vscode-extension",
-  "version": "0.0.3",
+  "version": "0.0.2",
   "lockfileVersion": 3,
   "requires": true,
   "packages": {
     "": {
       "name": "kani-vscode-extension",
-      "version": "0.0.3",
+      "version": "0.0.2",
```
Comment (qinheping) [inline package-lock.json:9]: Is this intended to revert the version number? 
If yes, we may want to have a separate PR for it.

Comment (jaisnan) [reply package-lock.json:9]: No, this is a mistake that is being corrected. It shouldnt have been 0.0.3 from the start so this is fixing it, but I can absolutely fix that in a seperate PR. Thanks for pointing that out.

File: src/debugger/debugger.ts [hunk 2]
```diff
@@ -59,19 +59,43 @@ async function getBinaryPath(): Promise<string | undefined> {
...
+	moduleName: string,
+	fileName: string,
+) {
 	// The binary that is being referred to here, is the binary present in the cargo artifacts.
 	// It looks like this - kani_concrete_playback_check_estimate_size_14615086421508420155
 	const binaryName = await getBinaryPath();
 
+	let functionParameter: string;
+
+	if (fileName === 'main') {
```
Comment (celinval) [inline src/debugger/debugger.ts:73]: What are you trying to do here? should you be checking if the function name is main instead?

File: src/debugger/debugger.ts [hunk 3]
```diff
@@ -59,19 +59,43 @@ async function getBinaryPath(): Promise<string | undefined> {
...
+		if (moduleName !== '') {
+			functionParameter = `${moduleName}::${functionName}`;
+		} else {
+			functionParameter = functionName;
+		}
+	} else if (fileName != '') {
+		if (moduleName === '') {
+			functionParameter = `${fileName}::${functionName}`;
```
Comment (celinval) [inline src/debugger/debugger.ts:81]: File name may not imply the module. For example, `mod.rs` usually implies that the module is the name of the folder where it is contained.

Another example is lib.rs.

File: src/model/runCargoTest.ts [hunk 4]
```diff
@@ -2,25 +2,42 @@
...
 	const taskDefinition: vscode.TaskDefinition = {
 		type: 'cargo',
 		command: 'test',
 		args: [functionName],
 	};
 
-	// Adding `--bin {packageName}` to the command prevents recompiling when
-	// there's no changes to the unit tests, which makes rerunning unit tests faster.
-	const cargoTestCommand: string = `RUSTFLAGS="--cfg=kani" cargo test --package ${packageName} --bin ${packageName} -- ${functionName} --exact --nocapture`;
+	if (fileName === 'main') {
```
Comment (celinval) [inline src/model/runCargoTest.ts:24]: this logic seems to be duplicated.

**Review by celinval — CHANGES_REQUESTED** [review]
I think the function derivation logic is a bit fragile and it has many corner cases which may not work.

File: src/debugger/debugger.ts (hunk 2 above)
Comment (jaisnan) [reply src/debugger/debugger.ts:73]: If the file name is main (or for main.rs), we don't need to pass the file name in the function parameter. In fact, passing it seems to break the unit test.

For example, in this example,

main.rs
``` rust
#[cfg(kani)]
mod verification {
    use super::*;

    #[kani::proof]
    pub fn stretched_rectangle_can_hold_original() {
        let original = Rectangle {
            width: kani::any(),
            height: kani::any(),
        };
        let factor = kani::any();
        kani::assume(0 != original.width);  //< explicit requirements
        kani::assume(0 != original.height); //< explicit requirements
        kani::assume(1 < factor); //< explicit requirements
        if let Some(larger) = original.stretch(factor) {
            assert!(larger.can_hold(&original));
        }
    }
    #[test]
    fn kani_concrete_playback_stretched_rectangle_can_hold_original_15788871033967399270() {
        let concrete_vals: Vec<Vec<u8>> = vec![
            // 4611686018427387904ul
            vec![0, 0, 0, 0, 0, 0, 0, 64],
            // 0ul
            vec![0, 0, 0, 0, 0, 0, 0, 0],
            // 2ul
            vec![2, 0, 0, 0, 0, 0, 0, 0],
        ];
        kani::concrete_playback_run(concrete_vals, stretched_rectangle_can_hold_original);
    }
}
```

the unit test name that needs to be passed is -> `verification::kani_concrete_playback_stretched_rectangle_can_hold_original_15788871033967399270`.

I was trying to create the parameter for such cases where the unit tests have a module but belong to main.rs

**Comment by jaisnan** [comment]
The solution seems unnecessary and the issue arises from issuing `--exact `in the first place into the test runner and debugger. Removing `--exact` fixed the issue so will be closing this PR. Thanks @celinval for the suggestion.

**Pull request closed without merging**